frontend/src/app/modules/user/teachers-page/teachers-page.component.ts

```diff
@@ -0,0 +1,95 @@
+import { DatePipe } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
+import { AuthService } from '@core/services/auth.service';
+import { LessonsCreateComponent } from '@modules/lessons/lessons-create/lessons-create.component';
+import { ModalComponent } from '@shared/components/modal/modal.component';
+import { IModal } from '@shared/models/IModal';
+import { IDateWithLessons } from '@shared/models/lesson/IDateWithLessons';
+import { UserShort } from '@shared/models/UserShort';
+
+import { TeacherStatistics } from 'src/app/models/lessons/teacher-statistics';
+import { LessonsService } from 'src/app/services/lessons.service';
+
+@Component({
+    selector: 'app-teachers-page',
+    templateUrl: './teachers-page.component.html',
+    styleUrls: ['./teachers-page.component.sass'],
+})
+export class TeachersPageComponent implements OnInit {
+    currentUser: UserShort = {
+        email: '',
+        firstName: '',
+        lastName: '',
+        imagePath: '',
+    };
+
+    statistics: TeacherStatistics = {
+        totalClasses: 0,
+        canceledClasses: 0,
+        futureClasses: 0,
+        totalStudents: 0,
+        nextClass: null,
+    };
+
+    datesWithLessons: IDateWithLessons[] = [];
+
+    constructor(
+        private authService: AuthService,
+        private dialogRef: MatDialog,
+        private lessonsService: LessonsService,
+    ) {}
+
+    ngOnInit(): void {
+        this.authService.loadUser().subscribe();
+
+        this.authService.user.subscribe((user) => {
+            this.currentUser = user;
+        });
+
+        this.lessonsService.getTeacherStatistics().subscribe((data) => {
+            this.statistics = data as TeacherStatistics;
+            if (this.statistics.nextClass) {
+                const datePipe = new DatePipe('en-US');
+
+                this.statistics.nextClass = datePipe.transform(this.statistics.nextClass, 'd MMMM yyyy, EEE');
```

Comment:
It's better to use pipe from html

Reply:
We import date string as well as numbers here, and date pipe converts everything to date string. We can't check typeoff of text in HTML, as it goes against the principle of separation of concerns. So to prevent unnecessary checks both in HTML and ts of lesson component, it's better to keep this without changes.
![image](https://user-images.githubusercontent.com/71650121/230777882-1e04676e-5467-48cf-84fe-001a67bc9c9a.png)